### PR TITLE
[Addition] Auto rebuild script

### PR DESCRIPTION
**Pull request by Vam-Jam**
Auto rebuild script as title says
Needs engine side PR merged for this to compile and work.

Script gets added via AddScript by localhost.as (which is called when using localhost.bat)
~~This can cause minor stutters, but we modders can disable/change how often its called, and we can work out how to improve it engine side later on.~~ read below

### CODE REVIEW

**Comment by Vam-Jam** [comment]
Added AutoRebuild.cfg file for people to edit.
```
enabled = true # bool - Script will be removed if AutoRebuild has been disabled.

check_x_tick = false# bool - Do we check to rebuild every x amount of ticks (can cause stutters)
check_timer = 90 # s32 - how often is x for rebuild tick check

check_focus = true # bool - Do we check when screen goes into focus
```
Added focus check, which will rebuild every time you alt tab back into the game.

By default focus check is enabled, and tick check is disabled.

File: Scripts/Solo/AutoRebuild.cfg [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Config settings for AutoRebuild script.
+
+enabled = true # bool - Script will be removed if AutoRebuild has been disabled.
+
+check_x_tick = false # bool - Do we check to rebuild every x amount of ticks (can cause stutters)
+check_timer = 90 # s32 - how often is x for rebuild tick check
```
Comment (eps0003) [inline Scripts/Solo/AutoRebuild.cfg:6]: I suggest removing check_x_tick and making it so check_timer = 0 disables it instead

File: Scripts/Solo/AutoRebuild.as [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	check_timer = cfg.read_s32("check_timer", 90);
+	check_focus = cfg.read_bool("check_focus", true);
+
+	if (!enabled)
+	{
+		removeMe(this);
+	}    
+}
+
+void onTick(CRules@ this)
```
Comment (eps0003) [inline Scripts/Solo/AutoRebuild.as:29]: Couldn't rebuild() be called twice on the same tick if both check_focus and check_x_tick are enabled? Could that be a problem?

Comment (Vam-Jam) [reply Scripts/Solo/AutoRebuild.as:29]: side from checking if we should rebuild twice this is not an issue

File: Scripts/Solo/AutoRebuild.cfg (hunk 1 above)
Comment (Vam-Jam) [reply Scripts/Solo/AutoRebuild.cfg]: done

**Review by AsuMagic — APPROVED** [review]

**Pull request merged**